test/e2e/parallel/space_test.go

```diff
@@ -77,6 +77,20 @@ func TestCreateSpace(t *testing.T) {
 		// then
 		VerifyResourcesProvisionedForSpace(t, awaitilities, space.Name)
 
+		// ensure that the space is reset back to the original (valid) target cluster
+		// so the cleanup logic can delete the Space
+		var resetOnce sync.Once
```

Comment:
Hm... what's the point of using `Once` here? resetOnce.Do() will be called a single time only (for this specific resetOnce instance) anyway. What am I missing?

Reply:
or were you were asking why do I define the variable `resetOnce` outside of the function? Well, the variable has to stay outside, so when the function is called, it always uses the same instance of the `Once`. If it was moved inside of the function, then it would create a new instance of `Once` for every call of the function

Reply:
should not be a problem - but should we move this definition in the parent test `TestCreateSpace` ( where we define `awaitilities` , `hostAwait`, `memberAwait` ) ? Just to be on the safe side.

Reply:
What I was missing is that the `reset()` was called twice (not once!). One in the defer and the second time in the sub-test bellow. I saw it only in called by defer and was wondering why do you need `once` if it's already called only once :)
Sorry for the confusion.

Reply:
> should we move this definition in the parent test TestCreateSpace ( where we define awaitilities , hostAwait, memberAwait ) ? Just to be on the safe side.

not really, it has to stay in the same (sub)test  where the space is provisioned, so it's tied to the actual space (and also the the function that executes this sub-test)